flake.nix

```diff
@@ -1,54 +1,93 @@
 {
   inputs = {
-    nixpkgs.url = "https://flakehub.com/f/NixOS/nixpkgs/0.2405.*";
-    rust-overlay = {
-      url = "https://flakehub.com/f/oxalica/rust-overlay/*";
+    nixpkgs.url = "https://flakehub.com/f/NixOS/nixpkgs/0.2411.*";
+
+    fenix = {
+      url = "https://flakehub.com/f/nix-community/fenix/0.1.*";
       inputs.nixpkgs.follows = "nixpkgs";
     };
-    crane.url = "https://flakehub.com/f/ipetkov/crane/0.19.*";
+
+    naersk.url = "https://flakehub.com/f/nix-community/naersk/0.1.*";
   };
 
   outputs = { self, ... }@inputs:
     let
+      lastModifiedDate = self.lastModifiedDate or self.lastModified or "19700101";
+      version = "${builtins.substring 0 8 lastModifiedDate}-${self.shortRev or "dirty"}";
+
       supportedSystems = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
-      forAllSystems = f: inputs.nixpkgs.lib.genAttrs supportedSystems (system: f rec {
+
+      forSystems = s: f: inputs.nixpkgs.lib.genAttrs s (system: f rec {
         pkgs = import inputs.nixpkgs {
           inherit system;
-          overlays = [
-            inputs.rust-overlay.overlays.default
-          ];
-        };
-
-        cranePkgs = pkgs.callPackage ./crane.nix {
-          inherit (inputs) crane;
-          inherit supportedSystems;
+          overlays = [ self.overlays.default ];
         };
       });
+
+      forAllSystems = forSystems supportedSystems;
     in
     {
-      packages = forAllSystems ({ cranePkgs, ... }: rec {
-        inherit (cranePkgs) flake-checker;
+      overlays.default = final: prev:
+        let
+          inherit (final.stdenv.hostPlatform) system;
+
+          rustToolchain = with inputs.fenix.packages.${system};
+            combine ([
+              stable.clippy
+              stable.rustc
+              stable.cargo
+              stable.rustfmt
+              stable.rust-src
+            ] ++ inputs.nixpkgs.lib.optionals (system == "x86_64-linux") [
+              targets.x86_64-unknown-linux-musl.stable.rust-std
+            ] ++ inputs.nixpkgs.lib.optionals (system == "aarch64-linux") [
+              targets.aarch64-unknown-linux-musl.stable.rust-std
+            ]);
+        in
+        {
+          inherit rustToolchain;
+
+          naerskLib = final.callPackage inputs.naersk {
+            cargo = rustToolchain;
+            rustc = rustToolchain;
+          };
+        };
+
+      packages = forAllSystems ({ pkgs, ... }: rec {
         default = flake-checker;
+
+        flake-checker = pkgs.naerskLib.buildPackage {
+          name = "flake-checker-${version}";
+          src = self;
+          doCheck = true;
+          buildInputs = with pkgs; [ ] ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [ Security SystemConfiguration ]);
+          nativeBuildInputs = with pkgs; [ ] ++ lib.optionals stdenv.isDarwin [ libiconv ];
+
+          env = {
```

Comment:
This looks wrong to me. I think you want these envs instead:

https://github.com/DeterminateSystems/flakehub-push/blob/d6b01371c58a1749effe44bc25d5ccf232416d1e/flake.nix#L66-L69

Reply:
Sadly this is the kind of thing you don't learn if you're full-time macOS heh...